kategory/src/main/kotlin/kategory/typeclasses/Composed.kt

```diff
@@ -0,0 +1,121 @@
+package kategory
+
+/**
+ * https://www.youtube.com/watch?v=wvSP5qYiz4Y
+ */
+interface ComposedType<out F, out G>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<F, HK<G, A>>.lift(): HK<ComposedType<F, G>, A> = this as HK<ComposedType<F, G>, A>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<ComposedType<F, G>, A>.lower(): HK<F, HK<G, A>> = this as HK<F, HK<G, A>>
+
+interface ComposedFoldable<in F, in G> :
+        Foldable<ComposedType<F, G>> {
+
+    fun FF(): Foldable<F>
+
+    fun GF(): Foldable<G>
+
+    override fun <A, B> foldL(fa: HK<ComposedType<F, G>, A>, b: B, f: (B, A) -> B): B =
+            FF().foldL(fa.lower(), b, { bb, aa -> GF().foldL(aa, bb, f) })
+
+    fun <A, B> foldLC(fa: HK<F, HK<G, A>>, b: B, f: (B, A) -> B): B = foldL(fa.lift(), b, f)
```

Comment:
The rule has an autofix, so it enforces consistency. Sadly it doesn't respect line size, nor knows how to break. The detekt team has been notified, no fix in sight. We can always revert it, remove the baseline to do a pass on lines that are too long, and re-apply the rule once once it's fixed.